File: gang/week16/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1; // 자기 자신은 포함되어 있어서 1로 시작
        for(int i = 1; i<=n; i++) {
            int sum = i;    // 시작하는 값을 i로 주고
            for(int j = i+1; j<=n; j++) {
                sum += j;   // i의 다음 값을 하나씩 더해서
                if(sum == n) {  // n이 완성되면 카운트
                    answer++;
                    break;
                }
                else if(sum > n) {  // 값이 n보다 커지면 이미 의미가 없기 때문에 break
                    break;
                }

            }
        }
        return answer;
    }
}

File: jeonghyeon/16week/숫자의 표현.js
function solution(n) {
    var answer = 0;
    for(let i = 1; i <= n; i++) {
        let sum = i; // 현재 더해진 수
        for(let j = i + 1; sum < n; j++) {
            sum += j; // 연속된 수 더해감
        }
        if(sum === n) {
            answer++; // n과 같은 수가 되면 증가
        }
    }
    return answer;
}